exonum/src/node/mod.rs

```diff
@@ -640,6 +640,20 @@ impl NodeHandler {
         self.add_timeout(timeout, time);
     }
 
+    fn maybe_add_propose_timeout(&mut self) {
+        if self.need_faster_propose() && self.allow_expedited_propose {
```

Comment:
I'd suggest to check for `self.allow_expedited_propose` at first, since rust does implement short circuit evaluation.